frontend/src/metabase/admin/settings/components/Email/BaseSMTPConnectionForm.tsx

```diff
@@ -0,0 +1,332 @@
+}
+
+const anySchema = Yup.mixed().nullable().default(null);
+
+const getFormValueSchema = (
+  settingsDetails: BaseSMTPConnectionFormProps["settingsDetails"],
+  secureMode?: boolean,
+) => {
+  const portSchema = secureMode
+    ? Yup.string()
+        .oneOf(["465", "587", "2525"], "Must be either 465, 587 or 2525")
+        .nullable()
+        .default("465")
+    : Yup.number()
+        .positive()
+        .nullable()
+        .required(Errors.required)
+        .default(null);
+
+  const securitySchema = secureMode
+    ? Yup.string()
+        .oneOf(
+          ["ssl", "tls", "starttls"],
+          "Must be either SSL, TLS or STARTTLS",
+        )
+        .nullable()
+        .default("ssl")
+    : Yup.string().default("none");
+
+  return Yup.object({
+    host: settingsDetails?.host?.is_env_setting
+      ? anySchema
+      : Yup.string().required(Errors.required).default(""),
+    port: settingsDetails?.port?.is_env_setting ? anySchema : portSchema,
+    security: settingsDetails?.security?.is_env_setting
+      ? anySchema
+      : securitySchema,
+    username: settingsDetails?.username?.is_env_setting
+      ? anySchema
+      : Yup.string().default(""),
+    password: settingsDetails?.password?.is_env_setting
+      ? anySchema
+      : Yup.string().default(""),
+  });
+};
+
+export const BaseSMTPConnectionForm = ({
+  onClose,
+  settingValues,
+  settingsDetails,
+  secureMode,
+  updateMutation,
+  deleteMutation,
+  dataTestId,
+  onTrackSuccess,
+}: BaseSMTPConnectionFormProps) => {
+  const [sendToast] = useToast();
+
+  const initialValues = useMemo(
+    () => ({
+      host: settingValues?.host ?? "",
+      port: settingValues?.port ?? (secureMode ? "465" : null),
+      security: settingValues?.security ?? (secureMode ? "ssl" : "none"),
+      username: settingValues?.username ?? "",
+      password: settingValues?.password ?? "",
+    }),
+    [settingValues, secureMode],
+  );
+
+  const handleClearEmailSettings = useCallback(async () => {
+    const result = await deleteMutation();
+    if (result.error) {
+      sendToast({
+        icon: "warning",
+        toastColor: "error",
+        message: isErrorWithMessage(result.error)
+          ? result.error.data.message
+          : t`Error clearing email settings`,
+      });
+    } else {
+      sendToast({
+        message: t`Email settings cleared`,
+      });
+    }
+  }, [deleteMutation, sendToast]);
+
+  const handleUpdateEmailSettings = useCallback(
+    async (formData: any) => {
+      try {
+        await updateMutation(formData).unwrap();
+        onTrackSuccess();
+        sendToast({
+          message: t`Email settings updated`,
+        });
+        onClose();
+      } catch (error) {
+        sendToast({
+          icon: "warning",
+          toastColor: "error",
+          message: getErrorMessage(error, t`Error updating email settings`),
+        });
+
+        // throw to allow the form to handle the error
+        throw error;
```

Comment:
I think this catch is preventing the line item errors from showing up in the UI?
<img width="428" height="110" alt="Screenshot 2025-07-10 at 4 52 13 PM" src="https://github.com/user-attachments/assets/8e93fc2a-0f13-4ead-a982-cbad4a702761" />